src/Utility/HttpStatus.php

```diff
@@ -0,0 +1,824 @@
+<?php
+/**
+ * Requests for PHP, an HTTP library.
+ *
+ * @copyright 2012-2023 Requests Contributors
+ * @license   https://github.com/WordPress/Requests/blob/stable/LICENSE ISC
+ * @link      https://github.com/WordPress/Requests
+ */
+
+namespace WpOrg\Requests\Utility;
+
+use WpOrg\Requests\Exception\InvalidArgument;
+
+/**
+ * Helper class for dealing with HTTP status codes.
+ *
+ * The codes are synced with the mdn web docs at https://developer.mozilla.org/en-US/docs/Web/HTTP/Status.
+ *
+ * @package Requests\Utilities
+ * @since   2.1.0
+ */
+final class HttpStatus {
+
+	// Informational responses (100 - 199)
+
+	/**
+	 * HTTP response code 100 - Continue.
+	 *
+	 * This interim response indicates that the client should continue the request or ignore the response if the request is already finished.
+	 *
+	 * @var string
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/100
+	 */
+	const TEXT_100 = 'Continue';
+
+	/**
+	 * HTTP response code 101 - Switching Protocols.
+	 *
+	 * This code is sent in response to an Upgrade request header from the client, and indicates the protocol the server is switching to.
+	 *
+	 * @var string
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/101
+	 */
+	const TEXT_101 = 'Switching Protocols';
+
+	/**
+	 * HTTP response code 102 - Processing (WebDAV).
+	 *
+	 * This code indicates that the server has received and is processing the request, but no response is available yet.
+	 *
+	 * @var string
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/102
+	 */
+	const TEXT_102 = 'Processing';
+
+	/**
+	 * HTTP response code 103 - Early Hints.
+	 *
+	 * This status code is primarily intended to be used with the Link header, letting the user agent start preloading resources while the server prepares a response or preconnects to an origin from which the page will need resources.
```

Comment:
Nitpick:
```suggestion
	 * This status code is primarily intended to be used with the Link header, letting the user agent
	 * start preloading resources while the server prepares a response or preconnects to an origin
	 * from which the page will need resources.
```

Here and in a few of the other places where the doclines are very long/wrapping.